third_party/terraform/resources/resource_google_project_services.go

```diff
@@ -90,10 +91,33 @@ func resourceGoogleProjectServicesRead(d *schema.ResourceData, meta interface{})
 	if err != nil {
 		return err
 	}
+
+	// use old services to set the correct renamed service names in state
+	s, _ := expandServiceUsageProjectServicesServices(d.Get("services"), d, config)
+	log.Printf("[DEBUG] Saw services in state on Read: %s ", s)
+	sset := golangSetFromStringSlice(s)
+	for ov, nv := range renamedServices {
+		_, ook := sset[ov]
+		_, nok := sset[nv]
+
+		// preserve any values set in prior state. If none were set, only store
+		// the old value.
+		if ook && nok {
+			continue
+		} else if ook {
+			delete(enabledSet, nv)
+		} else if nok {
+			delete(enabledSet, ov)
+		} else {
+			delete(enabledSet, nv)
```

Comment:
If the service is enabled upstream, list will return both {old, new} as set. In that case, we delete the new value so that only a single value is set. If it's not enabled, this will be a noop and neither will be set.